### PR TITLE
Use altitude mapping function from base class

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

**Comment by rmackay9** [comment]
LGTM.  probably should be rebased on master because the 1st commit has just gone in.

We should ask @jaxxzer to give his OK before merging.

File: Tools/autotest/ardusub.py [hunk 1]
```diff
@@ -164,6 +164,50 @@ def test_gripper_mission(self):
...
+        lat = 5
+        lon = 5
+        alt = 10
+
+        tstart = self.get_sim_time()
+        while True:
+            if self.get_sim_time() - tstart > 200:
+                raise NotAchievedException("Did not move far enough")
+            # send a position-control command
+            self.mav.mav.set_position_target_global_int_send(
```
Comment (jaxxzer) [inline Tools/autotest/ardusub.py]: Should we be sending this repeatedly, or just once before the loop?

File: ArduSub/GCS_Mavlink.cpp [hunk 2]
```diff
@@ -956,29 +946,20 @@ void GCS_MAVLINK_Sub::handleMessage(mavlink_message_t* msg)
...
-            case MAV_FRAME_GLOBAL_RELATIVE_ALT_INT:
-                loc.relative_alt = true;
-                loc.terrain_alt = false;
-                break;
-            case MAV_FRAME_GLOBAL_TERRAIN_ALT:
-            case MAV_FRAME_GLOBAL_TERRAIN_ALT_INT:
-                loc.relative_alt = true;
-                loc.terrain_alt = true;
+            Location::ALT_FRAME frame;
+            if (!mavlink_coordinate_frame_to_location_alt_frame(packet.coordinate_frame, frame)) {
```
Comment (jaxxzer) [inline ArduSub/GCS_Mavlink.cpp]: good, ty.

File: Tools/autotest/ardusub.py (hunk 1 above)
Comment (peterbarker) [reply Tools/autotest/ardusub.py]: guided mode times movements out if it doesn't receive stuff often enough (GUIDED_POSVEL_TIMEOUT_MS).

**Pull request merged**